sys/include/net/netopt.h

```diff
@@ -269,6 +269,18 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /**
+     * @brief   Type for sequiential processing of IPv6 extension headers
+     *
+     * On GET it requests a thread to provide an extension header.
+     * On SET it always returns -ENOTSUP.
+     *
+     * When the header is done the thread replies to the call.
+     * The result is the pktsnip_t with the extension header
+     * The result is NULL if no extension header is passed.
```

Comment:
You hand over a pointer (not a pointer of a pointer) below, so this should be impossible. Why not use the numeric return-value of NETAPI_GET (which is returned by `gnrc_netapi_get()`)?, e.g. `sizeof(gnrc_pktsnip_t)` on success, `-ENOMEM` (or whatever is applicable) on error.